src/super_gradients/training/datasets/detection_datasets/pascal_voc_detection.py

```diff
@@ -0,0 +1,140 @@
+import os
+import glob
+from pathlib import Path
+from typing import List, Tuple
+from xml.etree import ElementTree
+from tqdm import tqdm
+
+import numpy as np
+
+from super_gradients.training.utils.utils import download_and_untar_from_url, get_image_size_from_path
+from super_gradients.training.datasets.detection_datasets.detection_dataset import DetectionDataset
+from super_gradients.training.utils.detection_utils import DetectionTargetsFormat
+from super_gradients.common.abstractions.abstract_logger import get_logger
+from super_gradients.training.datasets.datasets_conf import PASCAL_VOC_2012_CLASSES_LIST
+
+logger = get_logger(__name__)
+
+
+class PascalVOCDetectionDataset(DetectionDataset):
+    """Dataset for Pascal VOC object detection"""
+
+    def __init__(self, data_dir: str, images_sub_directory: str, *args, **kwargs):
+        """Dataset for Pascal VOC object detection
+
+        :param data_dir:                Where the data is stored
+        :param images_sub_directory:    Sub directory of data_dir that includes images.
+        """
+        self.data_dir = data_dir
+        if not Path(data_dir).exists():
+            raise FileNotFoundError(f"Please make sure to download the data in the data directory ({self.data_dir}).")
```

Comment:
Why not raise this in parent class (obviously with the appropriate message, but if it does not exist we should raise it anyways)?

Reply:
Changed in b6e26294dec0e4bd447938aeb297500abbedc0ae